packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -513,4 +513,66 @@ void main() {
     expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 1.0);
   });
 
+  testWidgets('Switch renders correctly before, during, and after being tapped', (WidgetTester tester) async {
+    final Key switchKey = UniqueKey();
+    bool value = false;
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Center(
+              child: RepaintBoundary(
+                child: CupertinoSwitch(
+                  key: switchKey,
+                  value: value,
+                  dragStartBehavior: DragStartBehavior.down,
+                  onChanged: (bool newValue) {
+                    setState(() {
+                      value = newValue;
+                    });
+                  },
+                )
+              )
+            );
+          },
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byKey(switchKey),
+      matchesGoldenFile(
+        'switch.tap.off.png',
+        version: 0,
+      ),
+      skip: !isLinux,
+    );
+
+    await tester.tap(find.byKey(switchKey));
+    expect(value, isTrue);
+
+    // Kick off animation, then advance to intermediate frame.
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 60));
+    await expectLater(
+      find.byKey(switchKey),
+      matchesGoldenFile(
+        'switch.tap.turningOn.png',
+        version: 0,
+      ),
+      skip: !isLinux,
+    );  
```

Comment:
> ⏩ RUNNING: cd .; grep --line-number --extended-regexp [[:blank:]]$ packages/flutter/lib/src/cupertino/switch.dart packages/flutter/test/cupertino/switch_test.dart
packages/flutter/test/cupertino/switch_test.dart:565:    );  
🕐 ELAPSED TIME: 0:00:00.019169 for grep --line-number --extended-regexp [[:blank:]]$ 
packages/flutter/lib/src/cupertino/switch.dart packages/flutter/test/cupertino/switch_test.dart in .: 
Whitespace detected at the end of source code lines.
Please remove:

Looks like there're trailing white spaces at the end of this line. Could you remove it?

```suggestion
    );
```